### PR TITLE
Added missing AsTransaction* and code cleanup

### PR DESCRIPTION
**Pull request by albyrock87**
- Added missing AsTransaction* internal calls
- Added missing FilteredCollection transactions support
- Fixed compilation warnings
- Support newer feature version of SDKs

### CODE REVIEW

**Comment by albyrock87** [comment]
@PascalSenn could you review this PR? Or do you know who's the maintainer here? Thanks

**Review by PascalSenn — APPROVED** [review]
Thanks for the PR @albyrock87.  Sorry didnt see it

As noted, i do not think the AgressiveInlining makes any difference here. I mean, the method is still inlined, but i doubt that there will be any benefit in performance. 

 @glucaci  Do you want to leave the inlining in? If so this is good for me and can be merged.

File: src/Transactions/MongoTransactionClient.cs [hunk 1]
```diff
@@ -267,23 +268,24 @@ public IAsyncCursor<string> ListDatabaseNames(CancellationToken cancellationToke
...
         {
-            return _client.WithWriteConcern(writeConcern);
+            return _client.WithWriteConcern(writeConcern).AsTransactionClient();
         }
 
         public ICluster Cluster => _client.Cluster;
 
         public MongoClientSettings Settings => _client.Settings;
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
[the remark this answers is not in the document]
Comment (nscheibe) [reply src/Transactions/MongoTransactionClient.cs:288]: For me okay, usually I trust the JIT compiler...

**Review by nscheibe — APPROVED** [review]
Well done

**Pull request merged**